google-cloud-apis/src/com/google/cloud/tools/intellij/cloudapis/AddCloudLibrariesAction.java

```diff
@@ -48,52 +37,8 @@ public AddCloudLibrariesAction() {
   @Override
   @SuppressWarnings("MissingCasesInEnumSwitch")
   public void update(AnActionEvent e) {
-    boolean addLibrariesEnabled = false;
-    if (e.getProject() != null) {
-      Set<CloudLibrariesModuleSupportType> moduleSupportTypes =
-          Stream.of(ModuleManager.getInstance(e.getProject()).getModules())
-              .map(this::checkModuleForAddCloudLibraries)
-              .collect(Collectors.toSet());
-
-      addLibrariesEnabled = moduleSupportTypes.contains(CloudLibrariesModuleSupportType.SUPPORTED);
-      if (!addLibrariesEnabled) {
-        // update message to hint what is missing.
-        for (CloudLibrariesModuleSupportType supportType : moduleSupportTypes) {
-          switch (supportType) {
-            case MAVEN_REQUIRED:
-              e.getPresentation()
-                  .setDescription(
-                      GoogleCloudApisMessageBundle.message(
-                          "cloud.libraries.menu.action.maven.required.description"));
-              e.getPresentation()
-                  .setText(
-                      GoogleCloudApisMessageBundle.message(
-                          "cloud.libraries.menu.action.disabled.maven.text"));
-              break;
-            case APPENGINE_JAVA8_REQUIRED:
-              e.getPresentation()
-                  .setDescription(
-                      GoogleCloudApisMessageBundle.message(
-                          "cloud.libraries.menu.action.gae.java8.required.description"));
-              e.getPresentation()
-                  .setText(
-                      GoogleCloudApisMessageBundle.message(
-                          "cloud.libraries.menu.action.disabled.java8.text"));
-              break;
-          }
-        }
-      } else {
-        // standard message for a supported module action.
-        e.getPresentation()
-            .setDescription(
-                GoogleCloudApisMessageBundle.message("cloud.libraries.menu.action.description"));
-        e.getPresentation()
-            .setText(
-                GoogleCloudApisMessageBundle.message("cloud.libraries.menu.action.active.text"));
-      }
-    }
-
-    e.getPresentation().setEnabled(addLibrariesEnabled);
+    Stream.of(CloudApiActionDecoratorExtension.EP_NAME.getExtensions())
```

Comment:
You're right, it would be better to return something from this method and stop once one of the extensions do something and disables the action. For example, `decorate` could return `boolean` and if it's true, meaning action has been modified, other decorator extensions are not called then.